src/main/java/at/ac/fhcampuswien/richAF/services/OllamaService.java

```diff
@@ -0,0 +1,93 @@
+package at.ac.fhcampuswien.richAF.services;
+
+import at.ac.fhcampuswien.richAF.model.Config;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CompletableFuture;
+import org.json.JSONObject;
+/**
+ * Class containing the Ollama Service and its API Call
+ * @author Stefan
+ */
+public class OllamaService {
+    HttpClient client;
+    URI baseUri;
+    Config _config;
+
+    /**
+     * the first part prompt which will be sent which the API call, the prompt can be set with the Setter
+     * the first part contains the question to the LLM like "how many words are in the following text:"
+     * the second part, the text which shall get analyzed will get to the prompt with the call of the method
+     * @code askOllama
+     */
+    private String prompt;
+
+    public String getPrompt() {
+        return prompt;
+    }
+    public void setPrompt(String prompt) {
+        this.prompt = prompt;
+    }
+
+    /**
+     * Constructor:
+     * sets the local config file which will be needed in the api call
+     * creates a httpclient
+     * sets the endpoint url of the ollama api
+     * @param config
+     */
+    public OllamaService(Config config) {
+        _config = config;
+        client = HttpClient.newHttpClient();
+        baseUri = URI.create(_config.getProperty("ollama.endpoint"));
+    }
+
+    /**
+     * Sets the prompt to a standard prompt which was tested and provided the good results, the companyname is inserted in the text
+     * @param companyname
+     */
+    public void SetBasePrompt(String companyname){
+        prompt = "count the positive and negative news about the company "+companyname+" in the following text and return as result only just the count in exact this Format {positve=#;negative=#} instead of the # then the corresponding count number no text of the news in the response: ";
```

Comment:
Wenn es sich ausgeht wäre es nice, wenn die Prompts modular implementiert werden könnten. Wie bei den Logging configs könnte man diese in ein config file schreiben. Ist aber absolut kein muss, nur ein nice to have.

Reply:
sehr gute Idee, das werd ich noch ändern